tests/path.go

```diff
@@ -141,7 +139,7 @@ func (tp *TestSuite) TestInvalidPathRemainder(t *testing.T) {
 	}
 
 	_, err = api.ResolvePath(ctx, path.New("/ipld/"+nd.Cid().String()+"/bar/baz"))
-	if err == nil || !errors.As(err, &resolver.ErrNoLink{}) {
+	if err == nil || !strings.Contains(err.Error(), `no link named "bar"`) {
```

Comment:
We have to do this because there are HTTP API implementations of this interface (i.e.https://github.com/ipfs/go-ipfs-http-client) and we don't send the full type information over the wire.

So for now we're stuck with string checking the errors.